src/deluge/memory/cache_manager.cpp

```diff
@@ -104,7 +107,7 @@ uint32_t CacheManager::ReclaimMemory(MemoryRegion& region, int32_t totalSizeNeed
 			}
 
 			// Ok, we've got one Stealable
-			uint32_t* __restrict__ header = (uint32_t*)((uint32_t)stealable - 4);
+			auto* __restrict__ header = std::bit_cast<uint32_t*>((uint32_t)stealable - 4);
```

Comment:
If we're changing this anyway can we make it uintptr_t? No worries if that causes follow on issues

Reply:
Fixed